src/meal/models.py

```diff
@@ -114,12 +114,16 @@ class Component_ingredient(models.Model):
     component = models.ForeignKey(
         'meal.Component',
         verbose_name=_('component'),
-        related_name='+')
+        related_name='+',
+        on_delete=models.CASCADE
+    )
 
     ingredient = models.ForeignKey(
         'meal.Ingredient',
         verbose_name=_('ingredient'),
-        related_name='+')
+        related_name='+',
+        on_delete=models.CASCADE
```

Comment:
This is in the m2m through model `Component_ingredient`. It means a component can have multiple ingredients.

I would say a models.CASCADE on component and a models.PROTECT on ingredient. It means when we delete a component, we automatically delete all related Component_ingredient objects as the relationship will not exist anymore. But on the other side, when we delete an ingredient, we may want to protect it because there are still components relying on this ingredient.

Still, my opinion, it needs discussion. I will label all other m2m. @manumilou